cmd/kaspawallet/daemon/server/sync.go

```diff
@@ -200,71 +189,49 @@ func (s *server) refreshExistingUTXOsWithLock() error {
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
-	return s.refreshExistingUTXOs()
+	return s.refreshUTXOs()
 }
 
-func (s *server) addEntryToUTXOSet(entry *appmessage.UTXOsByAddressesEntry, addressSet walletAddressSet) error {
-	outpoint, err := appmessage.RPCOutpointToDomainOutpoint(entry.Outpoint)
-	if err != nil {
-		return err
-	}
-
-	utxoEntry, err := appmessage.RPCUTXOEntryToUTXOEntry(entry.UTXOEntry)
-	if err != nil {
-		return err
-	}
-
-	address, ok := addressSet[entry.Address]
-	if !ok {
-		return errors.Errorf("Got result from address %s even though it wasn't requested", entry.Address)
-	}
-
-	s.insertUTXO(&walletUTXO{
-		Outpoint:  outpoint,
-		UTXOEntry: utxoEntry,
-		address:   address,
-	})
+// updateUTXOSet clears the current UTXO set, and re-fills it with the given entries
+func (s *server) updateUTXOSet(entries []*appmessage.UTXOsByAddressesEntry) error {
+	utxos := make([]*walletUTXO, len(entries))
 
-	return nil
-}
-
-// insertUTXO inserts the given utxo into s.utxosSortedByAmount, while keeping it sorted.
-func (s *server) insertUTXO(utxo *walletUTXO) {
-	s.utxosSortedByAmount = append(s.utxosSortedByAmount, utxo)
-	// bubble up the new UTXO to keep the UTXOs sorted by value
-	index := len(s.utxosSortedByAmount) - 1
-	for index > 0 && utxo.UTXOEntry.Amount() > s.utxosSortedByAmount[index-1].UTXOEntry.Amount() {
-		s.utxosSortedByAmount[index] = s.utxosSortedByAmount[index-1]
-		index--
-	}
-	s.utxosSortedByAmount[index] = utxo
-}
+	for i, entry := range entries {
+		outpoint, err := appmessage.RPCOutpointToDomainOutpoint(entry.Outpoint)
+		if err != nil {
+			return err
+		}
 
-func (s *server) refreshExistingUTXOs() error {
-	addressSet := make(walletAddressSet, len(s.utxosSortedByAmount))
-	for _, utxo := range s.utxosSortedByAmount {
-		addressString, err := s.walletAddressString(utxo.address)
+		utxoEntry, err := appmessage.RPCUTXOEntryToUTXOEntry(entry.UTXOEntry)
 		if err != nil {
 			return err
 		}
 
-		addressSet[addressString] = utxo.address
+		address, ok := s.addressSet[entry.Address]
+		if !ok {
+			return errors.Errorf("Got result from address %s even though it wasn't requested", entry.Address)
+		}
+		utxos[i] = &walletUTXO{
+			Outpoint:  outpoint,
+			UTXOEntry: utxoEntry,
+			address:   address,
+		}
 	}
 
-	getUTXOsByAddressesResponse, err := s.rpcClient.GetUTXOsByAddresses(addressSet.strings())
+	sort.Slice(utxos, func(i, j int) bool { return utxos[i].UTXOEntry.Amount() > utxos[j].UTXOEntry.Amount() })
```

Comment:
Maybe it's better to test it on testnet first with a very big wallet?

Reply:
`n log n` should be fine any way